File: MMORPGServer/Proto/CreateLuaProto/Mail_RequestProto.lua
--请求邮件列表
Mail_RequestProto = { ProtoCode = 17004 }

--这句是重定义元表的索引，就是说有了这句，这个才是一个类
Mail_RequestProto.__index = Mail_RequestProto;

function Mail_RequestProto.New()
    local self = { }; --初始化self
    setmetatable(self, Mail_RequestProto); --将self的元表设定为Class
    return self;
end


--发送协议
function Mail_RequestProto.SendProto(proto)

    local ms = CS.LuaHelper.Instance:CreateMemoryStream();
    ms:WriteUShort(proto.ProtoCode);


    CS.LuaHelper.Instance:SendProto(ms:ToArray());
    ms:Dispose();
end


--解析协议
function Mail_RequestProto.GetProto(buffer)

    local proto = Mail_RequestProto.New(); --实例化一个协议对象
    local ms = CS.LuaHelper.Instance:CreateMemoryStream(buffer);


    ms:Dispose();
    return proto;
end

File: MMORPGServer/Proto/CreateProto/Mail_RequestProto.cs
//===================================================
//作    者：崔炜斌
//创建时间：2019-12-13 19:24:52
//备    注：
//===================================================
using System.Collections;
using System.Collections.Generic;
using System;

/// <summary>
/// 请求邮件列表
/// </summary>
public struct Mail_RequestProto : IProto
{
    public ushort ProtoCode { get { return 17004; } }


    public byte[] ToArray()
    {
        using (MMO_MemoryStream ms = new MMO_MemoryStream())
        {
            ms.WriteUShort(ProtoCode);
            return ms.ToArray();
        }
    }

    public static Mail_RequestProto GetProto(byte[] buffer)
    {
        Mail_RequestProto proto = new Mail_RequestProto();
        using (MMO_MemoryStream ms = new MMO_MemoryStream(buffer))
        {
        }
        return proto;
    }
}